etna/transforms/encoders/mean_segment_encoder.py

```diff
@@ -46,9 +50,27 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         -------
         :
             result dataframe
+
+        Raises
+        ------
+        ValueError:
+            If transform isn't fitted.
+        ValueError:
+            If there are segments that weren't present during training.
         """
+        if self.global_means is None:
+            raise ValueError("The transform isn't fitted!")
+
+        segments = df.columns.get_level_values("segment").unique().tolist()
+        new_segments = set(segments) - self.global_means.keys()
+        if len(new_segments) > 0:
+            raise ValueError(
+                f"This transform can't process segments that weren't present on train data: {new_segments}"
+            )
+
         df = self.mean_encoder.transform(df)
-        segment = df.columns.get_level_values("segment").unique()[0]
+        segments = df.columns.get_level_values("segment").unique()
```

Comment:
Can we reuse here the `segments` variable from line 64?

Reply:
Yes, will be fixed.